cmd/zetaclientd/keygen_tss.go

```diff
@@ -7,58 +7,33 @@ import (
 	"fmt"
 	"time"
 
-	"github.com/cometbft/cometbft/crypto/secp256k1"
 	"github.com/rs/zerolog"
 	tsscommon "github.com/zeta-chain/go-tss/common"
 	"github.com/zeta-chain/go-tss/keygen"
-	"github.com/zeta-chain/go-tss/p2p"
+	"github.com/zeta-chain/go-tss/tss"
 	"golang.org/x/crypto/sha3"
 
 	"github.com/zeta-chain/zetacore/pkg/chains"
 	observertypes "github.com/zeta-chain/zetacore/x/observer/types"
+	"github.com/zeta-chain/zetacore/zetaclient/chains/interfaces"
 	"github.com/zeta-chain/zetacore/zetaclient/context"
 	"github.com/zeta-chain/zetacore/zetaclient/metrics"
 	mc "github.com/zeta-chain/zetacore/zetaclient/tss"
 	"github.com/zeta-chain/zetacore/zetaclient/zetacore"
 )
 
-func GenerateTss(
+// GenerateTSS generates a new TSS if keygen is set.
+// If a TSS was generated successfully in the past,and the keygen was successful, the function will return without doing anything.
+// If a keygen has been set the functions will wait for the correct block to arrive and generate a new TSS.
+// In case of a successful keygen a TSS success vote is broadcasted to zetacore and the newly generate TSS is tested. The generated keyshares are stored in the correct directory
+// In case of a failed keygen a TSS failed vote is broadcasted to zetacore.
+func GenerateTSS(
 	appContext *context.AppContext,
 	logger zerolog.Logger,
-	client *zetacore.Client,
-	peers p2p.AddrList,
-	priKey secp256k1.PrivKey,
-	ts *metrics.TelemetryServer,
-	tssHistoricalList []observertypes.TSS,
-	tssPassword string,
-	hotkeyPassword string) (*mc.TSS, error) {
+	zetaCoreClient *zetacore.Client,
```

Comment:
I think appContext already has zetaCoreClient

Reply:
Not sure what you mean , could you please share a link ?